tests/Unit/Parallel/Actions/Test_Goto.cpp

```diff
@@ -49,4 +92,20 @@ SPECTRE_TEST_CASE("Unit.Parallel.GotoAction", "[Unit][Parallel][Actions]") {
   runner.force_next_action_to_be<component, Actions::Goto<Label2>>(0);
   runner.next_action<component>(0);
   CHECK(runner.get_next_action_index<component>(0) == 1);
+
+  runner.set_phase(Metavariables::Phase::TestRepeatUntil);
+  while (not ActionTesting::get_terminate<component>(runner, 0)) {
+    runner.next_action<component>(0);
+  }
+  CHECK(ActionTesting::get_databox_tag<component, HasConverged>(runner, 0));
+  CHECK(ActionTesting::get_databox_tag<component, Counter>(runner, 0) == 2);
+
+  // Test zero iterations of the `RepeatUntil` loop
+  runner.set_phase(Metavariables::Phase::TestRepeatUntil);
+  runner.template algorithms<component>().at(0).set_terminate(false);
+  while (not ActionTesting::get_terminate<component>(runner, 0)) {
+    runner.next_action<component>(0);
```

Comment:
can you put a counter inside this loop to demonstrate that it is entered exactly once... (if I understand what this is doing)

Reply:
Hm, what `RepeatUntil` does internally should not matter for this test. It should only make sure that the `Increment` action is not executed at all if the condition is already fulfilled from the start. That's being `CHECK`ed on line 110, or do you think it's not sufficient?

Reply:
yes, but it is not obvious that the `RepeatUntil` action is called. If I comment out line 105, the test will still pass...

Reply:
But line 105 exists, so the RepeatUntil action must be called to set_terminate to true :)
How about I add a check that the while loop is called _at least_ once?

Reply:
I agree with Larry.

Reply:
@wthrowe my concern is that it is not obvious that the `next_action` call on 108 actually calls `RepeatUntil`. So if the implementation of ActionTesting is changed so that `RepeatUntil` is not called by 108, then without the additional CHECK on 115 the test will still pass even though `RepeatUntil` is not called.    And even this check doesn't really confirm that `RepeatUntil` is called.  Browsing  ActionTesting.hpp, I think what we need is a call to `force_next_action_to_be`  and to then call `next_action` instead of the code from lines 104-15

Reply:
@kidder please check the fixup. It implements what we discussed yesterday.